File: src/main/java/at/yawk/fiction/impl/fimfiction/SearchPageParser.java
package at.yawk.fiction.impl.fimfiction;

import at.yawk.fiction.Pageable;
import at.yawk.fiction.impl.PageParser;
import java.util.ArrayList;
import org.jsoup.nodes.Element;

/**
 * @author yawkat
 */
class SearchPageParser extends PageParser<Pageable.Page<FimStory>> {
    @Override
    protected Pageable.Page<FimStory> create() {
        return new Pageable.Page<>();
    }

    @Override
    protected void parse(Element root, Pageable.Page<FimStory> target) throws Exception {
        target.setEntries(new ArrayList<>());

        for (Element storyContainer : root.select(".story_container")) {
            FimStory story = new FimStory();
            getParser(StoryParser.class).parse(storyContainer, story);
            target.getEntries().add(story);
        }
    }
}

File: src/main/kotlin/at/yawk/fiction/impl/fimfiction/SearchPageParser.kt
/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */

package at.yawk.fiction.impl.fimfiction

import at.yawk.fiction.Pageable
import at.yawk.fiction.Story
import at.yawk.fiction.impl.PageParser
import org.jsoup.nodes.Element

/**
 * @author yawkat
 */
internal object SearchPageParser : PageParser<Pageable.Page<Story>>() {
    override fun parse(root: Element): Pageable.Page<Story> {
        return Pageable.Page(
                entries = root.select(".story_container").map { StoryParser.parse(it) }.toList(),
                pageCount = null,
                last = false
        )
    }
}
